source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.h

```diff
@@ -84,6 +124,9 @@ class LocalRateLimiterImpl {
   TokenState tokens_;
   absl::flat_hash_set<LocalDescriptorImpl, LocalDescriptorHash, LocalDescriptorEqual> descriptors_;
   std::vector<LocalDescriptorImpl> sorted_descriptors_;
+
+  ShareProviderSharedPtr share_provider_;
```

Comment:
OOC does this really need to be a shared-ptr and not just a reference. AFAIK the singleton should outlive the filter, but I may be mistaken.

Reply:
Yeah, the singleton has longer lifetime than the filter. But note the `getShareProvider()` not guarantee that the 
singleton will keep a copy of returned shared pointer.

For example, if we support new algorithms in the future, then the singleton will only keep a map from the proto to the weak pointer of share provider.